octoprint_mrbeam/analytics/analytics_handler.py

```diff
@@ -42,900 +40,635 @@ def existing_analyticsHandler():
 			except Exception as e:
-				self._logger.exception('Error during add_laser_intensity_value: {}'.format(e.message))
+				self._logger.exception('Exception during collect_laser_intensity_value: {}'.format(e))
 
-	def _write_dust_log(self, data):
-		try:
-			if self._analyticsOn:
-				self._write_jobevent(ak.FINAL_DUST, payload=data)
-		except Exception as e:
-			self._logger.exception('Error during write dust_log: {}'.format(e.message))
+	def _init_collectors(self):
+		self._current_dust_collector = ValueCollector('DustColl')
+		self._current_intensity_collector = ValueCollector('IntensityColl')
+		self._current_lasertemp_collector = ValueCollector('TempColl')
 
-	def write_fan_rpm_test(self, data):
-		try:
-			if self._analyticsOn:
-				self._write_jobevent(ak.FAN_RPM_TEST, payload=data)
-		except Exception as e:
-			self._logger.exception('Error during write_fan_50_test: {}'.format(e.message))
+	def _add_collector_details(self):
+		intensity_summary = self._current_intensity_collector.getSummary()
+		lasertemp_summary = self._current_lasertemp_collector.getSummary()
 
-	def _init_jsonfile(self):
-		open(self._jsonfile, 'w+').close()
-		data = {
-			ak.LASERHEAD_VERSION: self._getLaserHeadVersion(),
+		lh_info = {
+			ak.Device.LaserHead.VERSION: None,
+			ak.Device.LaserHead.SERIAL: self._laserhead_handler.get_current_used_lh_data()['serial'],
 		}
-		self._write_deviceinfo(ak.INIT, payload=data)
-		self._write_current_software_status()
 
-	def _write_new_line(self):
-		if self._analyticsOn:
-			try:
-				if not os.path.isfile(self._jsonfile):
-					self._init_jsonfile()
-				with open(self._jsonfile, 'a') as f:
-					f.write('\n')
-			except Exception as e:
-				self._logger.exception('Error while writing newline: {}'.format(e.message), analytics=False)
+		intensity_summary.update(lh_info)
+		lasertemp_summary.update(lh_info)
+
+		self._add_job_event(ak.Job.Event.Summary.DUST, payload=self._current_dust_collector.getSummary())
+		self._add_job_event(ak.Job.Event.Summary.INTENSITY, payload=intensity_summary)
+		self._add_job_event(ak.Job.Event.Summary.LASERTEMP, payload=lasertemp_summary)
+
+	# -------- JOB HELPER METHODS --------------------------------------------------------------------------------------
+	def _cleanup_job(self):
+		self._current_job_id = None
+		self._current_dust_collector = None
+		self._current_intensity_collector = None
+		self._current_lasertemp_collector = None
+		self._current_cpu_data = None
+		self._current_job_time_estimation = None
+
+	def _init_new_job(self):
+		self._cleanup_job()
+		self._current_job_id = 'j_{}_{}'.format(self._snr, time.time())
+		self._add_job_event(ak.Job.Event.LASERJOB_STARTED)
 
-	def _append_data_to_file(self, data):
-		if self._analyticsOn:
+	# -------- WRITER THREAD (queue --> analytics file) ----------------------------------------------------------------
+	def _write_queue_to_analytics_file(self):
+		while self._analytics_enabled:
 			try:
```

Comment:
bring the try/catch to the outer most level

Reply:
done